tests/FakeItEasy.Specs/AssignsOutAndRefParametersSpecs.cs

```diff
@@ -2,122 +2,175 @@ namespace FakeItEasy.Specs
 {
     using System;
     using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Configuration;
     using FakeItEasy.Tests.TestHelpers;
     using FluentAssertions;
     using Xbehave;
 
-    public interface IHaveAnOut
+    [SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", MessageId = "WithOut", Justification = "That's two words, not one")]
+    [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "1#", Justification = "Required for testing.")]
+    [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#", Justification = "Required for testing.")]
+    public delegate void VoidDelegateWithOutAndRefParameters(int byValueParameter, ref int byRefParameter, out int outParameter);
+
+    [SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", MessageId = "WithOut", Justification = "That's two words, not one")]
+    [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "1#", Justification = "Required for testing.")]
+    [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#", Justification = "Required for testing.")]
+    public delegate int NonVoidDelegateWithOutAndRefParameters(int byValueParameter, ref int byRefParameter, out int outParameter);
+
+    public interface IHaveARef
     {
-        [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "0#", Justification = "Required for testing.")]
-        void MightReturnAKnownValue(out string andThisIsWhoReallyDidIt);
+        [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "0#", Justification = "Required for testing.")]
+        void MightReturnAKnownValue(ref string andThisIsWhoReallyDidIt);
     }
 
     public static class AssignsOutAndRefParametersSpecs
     {
+        const string Condition = "someone_else";

```

Comment:
Late-breaking comment. I don't think we have a test where we establish a condition and then don't meet it when exercising the subject. If we care about the ability to decide whether or not to assign the output, then maybe we should have an extra test. 
If we don't, the tests could become simpler...